src/Data/String/CodePoints.purs

```diff
@@ -0,0 +1,273 @@
+-- | These functions allow PureScript strings to be treated as if they were
+-- | sequences of Unicode code points instead of their true underlying
+-- | implementation (sequences of UTF-16 code units). For nearly all uses of
+-- | strings, these functions should be preferred over the ones in Data.String.
+module Data.String.CodePoints
+  ( module StringReExports
+  , CodePoint()
+  , codePointAt
+  , codePointFromInt
+  , codePointToInt
+  , count
+  , drop
+  , dropWhile
+  , fromCodePointArray
+  , indexOf
+  , indexOf'
+  , lastIndexOf
+  , lastIndexOf'
+  , length
+  , singleton
+  , splitAt
+  , take
+  , takeWhile
+  , toCodePointArray
+  , uncons
+  ) where
+
+import Prelude
+
+import Data.Array as Array
+import Data.Char as Char
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.String as String
+import Data.String.Unsafe as Unsafe
+-- WARN: In order for this module to be a drop-in replacement for Data.String,
+-- this list must be updated to re-export any exports added to Data.String.
```

Comment:
Right; how about something like:

> If a new function is added to Data.String, a version of that function should be exported from this module, which should be the same except that it should operate on the code point level rather than the code unit level. If the function's behaviour does not change based on whether we consider strings as sequences of code points or code units, it can simply be re-exported from Data.String.